File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/ui/viewmodels/TaskListViewModel.java
package com.vladislavmyasnikov.courseproject.ui.viewmodels;

import android.app.Application;

import com.vladislavmyasnikov.courseproject.data.DataRepository;
import com.vladislavmyasnikov.courseproject.data.db.entity.TaskEntity;

import java.util.List;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MediatorLiveData;
import androidx.lifecycle.Observer;

public class TaskListViewModel extends AndroidViewModel {

    private DataRepository mDataRepository;
    private MediatorLiveData<List<TaskEntity>> mTasks = new MediatorLiveData<>();

    public TaskListViewModel(Application application) {
        super(application);
        mDataRepository = DataRepository.Companion.getInstance(application);
    }

    public void init(int lectureId) {
        mTasks.addSource(mDataRepository.loadTasks(lectureId), new Observer<List<TaskEntity>>() {
            @Override
            public void onChanged(List<TaskEntity> tasks) {
                mTasks.postValue(tasks);
            }
        });
    }

    public LiveData<List<TaskEntity>> getTasks() {
        return mTasks;
    }
}

File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/ui/viewmodels/TaskListViewModel.kt
package com.vladislavmyasnikov.courseproject.ui.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import com.vladislavmyasnikov.courseproject.data.db.entity.TaskEntity
import com.vladislavmyasnikov.courseproject.data.repositories.TaskRepository
import com.vladislavmyasnikov.courseproject.ui.main.App

class TaskListViewModel(application: Application) : AndroidViewModel(application) {

    private val taskRepository: TaskRepository by lazy {
        App.instance.repositoryComponent.getTaskRepository()
    }
    private val mLectureId = MutableLiveData<Int>()

    val tasks: LiveData<List<TaskEntity>> = Transformations.switchMap(mLectureId) {
        id -> taskRepository.getTasksByLectureId(id)
    }

    fun loadTasksByLectureId(id: Int) {
        mLectureId.value = id
    }
}
